src/elements/content-sidebar/activity-feed/annotation-thread/useAnnotationAPI.js

```diff
@@ -31,46 +33,78 @@ type UseAnnotationAPI = {
         message?: MessageDescriptor,
         title?: MessageDescriptor,
     },
+    handleCreateReply: (message: string) => void,
     handleDelete: ({ id: string, permissions: AnnotationPermission }) => any,
+    handleDeleteReply: ({ id: string, permissions: BoxCommentPermission }) => void,
     handleEdit: (id: string, text: string, permissions: AnnotationPermission) => void,
+    handleEditReply: (
+        replyId: string,
+        message: string,
+        status?: FeedItemStatus,
+        hasMention: boolean,
+        permissions: BoxCommentPermission,
+    ) => void,
     handleStatusChange: (id: string, status: FeedItemStatus, permissions: AnnotationPermission) => void,
     isLoading: boolean,
+    replies: Array<Comment>,
 };
 
-const useAnnotationAPI = ({ annotationId, api, fileId, filePermissions, errorCallback }: Props): UseAnnotationAPI => {
+const useAnnotationAPI = ({
+    annotationId,
+    api,
+    currentUser,
+    fileId,
+    filePermissions,
+    errorCallback,
+}: Props): UseAnnotationAPI => {
     const [annotation, setAnnotation] = React.useState();
+    const [replies, setReplies] = React.useState<{ [string]: Comment }>({});
     const [error, setError] = React.useState();
     const [isLoading, setIsLoading] = React.useState(true);
 
-    const getAnnotationSuccess = (fetchedAnnotation: Annotation): void => {
-        setAnnotation(fetchedAnnotation);
-        setError(undefined);
-        setIsLoading(false);
+    const normalizeReplies = (repliesArray: Array<Comment>): { [string]: Comment } => {
+        if (!repliesArray) {
+            return {};
+        }
+        return repliesArray.reduce((prevValues, reply) => {
+            return {
+                ...prevValues,
+                [reply.id]: reply,
+            };
+        }, {});
+    };
+
+    const denormalizeReplies = (repliesMap: { [string]: Comment }): Array<Comment> => {
+        return Object.keys(repliesMap).map(key => repliesMap[key]);
     };
 
+    const { handleCreateReply, handleEditReply, handleDeleteReply } = useRepliesAPI({
+        annotationId,
+        api,
+        currentUser,
+        fileId,
+        filePermissions,
+        setReplies,
```

Comment:
Can we encapsulate our state management by handling this via callbacks rather than passing the setter down? So, passing in something like a `handleReplyUpdate` handler?